### PR TITLE
#799 Custom filters

### PR DESCRIPTION
**Pull request by Mizzick**
#799

### CODE REVIEW

File: Extension/lib/filter/subscription.js [hunk 1]
```diff
@@ -108,6 +108,99 @@ adguard.subscriptions = (function (adguard) {
...
+
+            return result;
+        }
+
+        return {
+            name: parseTag('Title'),
+            description: parseTag('Description'),
+            homepage: parseTag('Homepage'),
+            version: parseTag('Version'),
+            timeUpdated: parseTag('TimeUpdated'),
```
Comment (atropnikov) [inline Extension/lib/filter/subscription.js]: `TimeUpdated` is our custom property.

File: Extension/lib/filter/subscription.js [hunk 2]
```diff
@@ -108,6 +108,99 @@ adguard.subscriptions = (function (adguard) {
...
+            expires: parseTag('Expires')
```
Comment (atropnikov) [inline Extension/lib/filter/subscription.js]: We should parse `Expires` properly.

File: Extension/lib/filter/subscription.js [hunk 3]
```diff
@@ -108,6 +108,99 @@ adguard.subscriptions = (function (adguard) {
...
+                var languages = [];
+                var displayNumber = 0;
+
+                filter = new SubscriptionFilter(filterId, groupId, defaultName, defaultDescription, homepage, version, timeUpdated, displayNumber, languages, expires, subscriptionUrl);
+                filter.loaded = true;
+                //custom filters have special field
+                filter.customUrl = url;
+
+                filters.push(filter);
+                filtersMap[filter.filterId] = filter;
```
Comment (atropnikov) [inline Extension/lib/filter/subscription.js]: You should save filter metadata to the storage.

File: Extension/lib/filter/subscription.js [hunk 4]
```diff
@@ -108,6 +108,99 @@ adguard.subscriptions = (function (adguard) {
...
+     * Adds or updates custom filter
+     *
+     * @param url subscriptionUrl
+     * @param callback
+     */
+    var updateCustomFilter = function (url, callback) {
+
+        adguard.backend.loadFilterRulesBySubscriptionUrl(url, function (rules) {
+            //Check if filter from this url was added before
+            var filter = filters.find(function (f) {
```
Comment (atropnikov) [inline Extension/lib/filter/subscription.js]: You don't check that the new version differs from the previous. So after updating it's always showing popup about updating.

**Comment by Mizzick** [comment]
it does not look like anything now, cause first we need to complete #798

**Comment by Mizzick** [comment]
screenshot 2017-08-10 15 32 51

**Pull request closed without merging**